lib/limitd_pool.js

```diff
@@ -6,15 +6,29 @@ const clients = new Map();
 exports.prepare = function(address) {
   const limitd = new LimitdClient({hosts: [address]});
 
-  limitd.on('close', () => {
-    clients.delete(address);
-  });
+  function addToMap() {
```

Comment:
can we pass the parameters? we have both of them, don't need the closure, right? otherwise, the call in line #31 does not have enough context